smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/SmackIntegrationTestFramework.java

```diff
@@ -392,6 +422,28 @@ private void runTests(Set<Class<? extends AbstractSmackIntTest>> classes)
                     it.remove();
                     continue;
                 }
+
+                final String specification;
+                if (method.isAnnotationPresent(SpecificationReference.class)) {
```

Comment:
Simply re-use `getSpecificationReference()` here

Reply:
This only gets the specification's document, not the entire thing. This entire block, however, should be moved up and operate on a class level, instead of a method level.